detekt-cli/build.gradle.kts

```
@@ -42,12 +42,18 @@ tasks {
     }
 
     val runWithHelpFlag by registering(JavaExec::class) {
+        inputs.files(shadowJar)
+        outputs.upToDateWhen { true }
         classpath = files(shadowJar)
         args = listOf("--help")
     }
 
     val runWithArgsFile by registering(JavaExec::class) {
-        inputs.files(formattingJar) // ensures detekt-formatting JAR is built before this task is executed
+        // The task generating these jar files run first.
+        inputs.files(shadowJar, formattingJar)
+        // This task does not adopt incremental-build (up-to-date) check because it is reading
+        // the entire directory as the input source.
+        outputs.upToDateWhen { false }
```

If outputs are added correctly then this line isn't necessary, and Gradle would check if the report outputs are up to date or not based on the inputs.

---

This `outputs.upToDateWhen { false }` is unnecessary - But I would want to explicitly put it here so that we do not need to look into further optimization

---

I just remembered there was recent functionality added to specifically mark a task as unsuitable for up to date checks or caching:

```suggestion
        doNotTrackState("The entire root directory is read as the input source.")
```

There's some more info in the [Gradle 7.3 release notes](https://docs.gradle.org/7.3/release-notes.html#plugin-development-improvements):

> @UntrackedTask and Task.doNotTrackState are a replacement for Task.outputs.upToDateWhen { false } if you want your task to never be up-to-date. It has the advantage that it is faster since Task.outputs.upToDateWhen { false } still spends time on capturing task state.